File: python/courtlistener/api/gifts.py
"""
Gifts API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class GiftsAPI(BaseAPI):
    """API for accessing gifts data from financial disclosures."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "gifts/"
    
    def list(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List gift records.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of gifts
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, gift_id: int) -> Dict[str, Any]:
        """
        Get a specific gift by ID.
        
        Args:
            gift_id: Gift ID
        
        Returns:
            Gift data
        """
        return self.client.get(f"{self.endpoint}{gift_id}/")
    
    def paginate(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated gifts.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for gifts
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/Gifts.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Gifts API endpoint
 */
class Gifts extends BaseApi
{
    protected string $endpoint = 'gifts/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List gifts with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listGifts(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific gifts by ID
     *
     * @param int|string $id Gifts ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getGifts($id, array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search gifts
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchGifts(array $params = []) {
        return $this->search($params);
    }
}
